oodles/core/classes/framework.py

```diff
@@ -39,166 +49,111 @@ def __init__(self, cfg={}):
-            extra_args=extra_args,
-        )
-        if not saved["is_repeat"]:
-            self.summary_data["all_data"].append(saved)
-            self.predicted_count += 1
-
-    def add_data_point_to_smart_warehouse(
-        self, inputs, outputs, gts=None, extra_args={}
-    ):
-        """Logs only the interesting test cases to data warehouse. Logged under sub-folder 'smart_data'"""
-
-        saved = self.add_data_point_to_warehouse(
-            inputs,
-            outputs,
-            "smart_data/",
-            gts=gts,
-            extra_args=extra_args,
-        )
-        if not saved["is_repeat"]:
-            self.summary_data["smart_data"].append(saved)
-            self.selected_count += 1
-
-    def add_data_point_to_warehouse(
-        self, inputs, outputs, warehouse, gts=None, extra_args={}
-    ):
-        """Creates a json file per data-point.
-        Deletes model-related fields and transforms input, output data in a json-compatible format.
-        """
-
-        if "model" in inputs:
-            del inputs["model"]
-
-        identifier = extra_args["identifier"]
-        path = (
-            self.fold_name
-            + "/"
-            + str(self.version)
-            + "/"
-            + warehouse
-            + str(identifier[0])
-            + ".json"
-        )
-        is_repeat = True
-
-        if not os.path.exists(path):
-            is_repeat = False
-            datapoint = {
-                "input": json.dumps(inputs, cls=NumpyEncoder),
-                "output": json.dumps(outputs, cls=NumpyEncoder),
-                "gt": json.dumps(gts, cls=NumpyEncoder),
-                "extra_args": json.dumps(extra_args, cls=NumpyEncoder),
-                "identifier": json.dumps(identifier, cls=NumpyEncoder),
-            }
-            write_json(
-                path,
-                datapoint,
+        if "golden_testing_dataset" in evaluation_args:
+            self.set_golden_testing_dataset(evaluation_args["golden_testing_dataset"])
+        if "training_func" in training_args:
+            self.set_training_func(training_args["training_func"])
+        if "inference_func" in evaluation_args:
+            self.set_inference_func(evaluation_args["inference_func"])
+
+    def reset_retraining(self):
+        self.version += 1
+        self.selected_count = 0
+        self.smart_data_ids = []
+        retrain_folder = os.path.join(self.fold_name, str(self.version))
+        if not os.path.exists(retrain_folder):
+            os.mkdir(retrain_folder)
+
+    def get_data_id(self, inputs):
+        if self.data_id_type in inputs:
+            return inputs[self.data_id_type]
+        elif self.data_id_type == "utc_timestamp":
+            timestamp = str(datetime.utcnow().timestamp()).replace(".", "")
+            rand_int = random.sample(
+                range(10 * self.batch_size), self.batch_size
+            ).sort()
+            ids = [int(timestamp + str(x)) for x in rand_int]
```

Comment:
You need to do (timestamp * 10000 + x)?  Else, elems of next batch can have ids < elems of current batch?